docs/source/daml/intro/5_Restrictions.rst

```diff
@@ -0,0 +1,183 @@
+- ``pass (days 10) : Scenario ()`` is a recipe for a transaction that doesn't submit any transactions, but has the side-effect of changing the LET of the test ledger. It returns ``()``, also called ``Unit`` and can be thought of as a zero-tuple.
+- ``create iou : Update (ContractId Iou)``, where ``iou : Iou`` is a recipe for a transaction consisting of a single ``create`` action, and returns the contract id of the created contract if successful.
+- ``submit alice (create iou) : Scenario (ContractId Iou)`` is a recipe for a scenario in which Alice evaluates the result of ``create iou`` to get a transaction and a return value of type ``ContractId Iou``, and then submits that transaction to the ledger.
+
+Any DAML ledger knows how to perform actions of type ``Update a``. Only some know how to run scenarions, meaning they can perform actions of type ``Scenario a``.
+
+An action followed by another action, possibly depending on the result of the first action, is just another action. Specifically:
+
+- A transacion is a list of actions so a transaction followed by another transaction is again a transaction.
+- A scenario is a list of interactions with the ledger (``submit``, ``getParty``, ``pass``, etc.) so a scenario followed by another scenario is again a scenario.
+
+This is where ``do`` blocks come in. ``do`` blocks allow you to combine small actions in to bigger actions, using the results of earlier actions in later ones.
+
+.. literalinclude:: daml/daml-intro-5/Restrictions.daml
+  :language: daml
+  :start-after: -- DO_DEMO_BEGIN
+  :end-before: -- DO_DEMO_END
+
+Above, we see ``do`` blocks in action for both ``Scenario`` and ``Update``.
+
+You may already have noticed the use of ``return`` in the redeem choice. ``return x`` is a no-op action which returns value ``x`` so ``return 42 : Update Int``. Since ``do`` blocks always return the value of their last action, ``sub_scenario2 : Scenario Int``.
+
+Failing Actions
+---------------
+
+Not only are ``Update`` and ``Scenario`` examples of ``Action``, they are both examples of actions that can fail, e.g. because a transaction is illegal or the party retrieved via ``getParty`` doesn't exist on the ledger.
+
+Each has a special action ``abort txt`` which takes on type ``Update ()`` or ``Scenario ()`` depending on context and represents failure. Transactions and scenarios succeed or fail *atomically* as a whole so an occurrence of an ``abort`` action will always fail the entire evaluation of the current ``Scenario`` or ``Update``.
+
+The last expression in the ``do`` block of the ``Redeem`` choice is a pattern matching expression on ``dow``. It has type ``Update ()`` and is either an ``abort`` or ``return`` depending on the day of week. So during the week, it's a no-op and on weekends, it's the special failure action. Thanks to the atomicity of transactions, no transaction can ever make use of the ``Redeem`` choice on weekends.
```

Comment:
```suggestion
The last expression in the ``do`` block of the ``Redeem`` choice is a pattern matching expression on ``dow``. It has type ``Update ()`` and is either an ``abort`` or ``return`` depending on the day of week. So during the week, it's a no-op and on weekends, it's the special failure action. Thanks to the atomicity of transactions, no transaction can ever make use of the ``Redeem`` choice on weekends, because it fails the entire transaction.
```